### ISSUE TITLE
Flaky gui test: test_claim_device_invitation_cancelled

### ISSUE THREAD

**Issue by AntoineDupre** [Scille/parsec-cloud#1362]
The ci sometimes fails with:
```
================================== FAILURES ===================================
________ test_claim_device_invitation_cancelled[step_2_start_greeter] _________
value = <trio.Nursery object at 0x000001A182D84B38>
    async def yield_(value=None):
>       return await _yield_(value)
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\async_generator\_impl.py:106: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\async_generator\_impl.py:99: in _yield_
    return (yield _wrap(value))
tests\core\gui\devices_widget\test_claim.py:533: in test_claim_device_invitation_cancelled
    await CancelledTestBed().run()
E         +2
E         -1
tests\core\gui\devices_widget\test_claim.py:463: AssertionError]
```

**Comment by AntoineDupre** [comment]
It looks like an `InvitationAlreadyUsedError` can be raised from `parsec/core/invite/claimer._do_wait_peer` instead of the expected `BackendNotAvailable`:
```
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\parsec\core\gui\claim_device_widget.py", line 65, in run
    in_progress_ctx = await initial_ctx.do_wait_peer()
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\parsec\core\invite\claimer.py", line 121, in do_wait_peer
    claimer_sas, greeter_sas, shared_secret_key = await self._do_wait_peer()
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\parsec\core\invite\claimer.py", line 80, in _do_wait_peer
    _check_rep(rep, step_name="step 1")
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\parsec\core\invite\claimer.py", line 40, in _check_rep
    raise InviteAlreadyUsedError
parsec.core.invite.exceptions.InviteAlreadyUsedError: Invitation already used
```

**Comment by AntoineDupre** [comment]
related issue #1365
closed by #1363 
closed by #1372
